ManagementPacks/Community.DataOnDemand.Unix/Scripts/ResolveAddress.pl

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/perl -w
+# Provides Provides name/IpAddress information in a format expected by Squared Up's Visual Application Discovery and Analysis feature.
+# Copyright 2016 Squared Up Limited, All Rights Reserved.
+
+use strict;
+use warnings;
+use Socket;
+
+# Handle arguments and global vars
+my $format = "csv";
+my $lineEnd;
+my @queries;
+
+if ( $#ARGV &lt; 0 || $#ARGV &gt; 1 ) {
+    print "Usage: $0 [format] nameOrIp\n";
+    exit(1);
+}
+elsif ($#ARGV == 0) {
+    @queries = split(/\s*,\s*/, $ARGV[0]);
+}
+elsif ($#ARGV == 1) {
+    $format = $ARGV[0];
+    @queries = split(/\s*,\s*/, $ARGV[1]);
+}
+
+# Set output format
+if ($format eq "csv") {
+    $lineEnd = "\n";
+}
+elsif ($format eq "csvEx") {
+    $lineEnd = "\%EOL\%"
+}
+else {
+    print "Unknown format '$format'\n";
+    exit(1);
+}
+
+# Although Socket could be used instead of regex and manual validation, the behaviour and availability is not platform consistent.
+sub _is_ipv4 {
+    shift if ref $_[0];
+    my $value = shift;
+
+    return undef unless defined($value);
+
+    my (@octets) = $value =~ /^(\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})$/;
```

Comment:
I feel like this line, along with 48 and 49 could do with a small comment just explaining roughly what their effects  [on each octet] are  (esp. given nobody really knows Perl)